initfiles/componentfiles/configschema/xsd/esp_service_wssql.xsd

```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+#    HPCC SYSTEMS software Copyright (C) 2018 HPCC Systems®.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+-->
+
+<xs:schema
+        xmlns:xs="http://www.w3.org/2001/XMLSchema" elementFormDefault="qualified" attributeFormDefault="unqualified"
+        xmlns:hpcc="someuri">
+    <xs:include schemaLocation="types.xsd"/>
+    <hpcc:insert hpcc:schemaPath="/Environment/Software">
+        <xs:element name="EspService" hpcc:displayName="WS SQL Service" hpcc:itemType="wssql" minOccurs="0"
+                    maxOccurs="unbounded" hpcc:class="component">
+            <xs:complexType>
+                <xs:attributeGroup ref="buildInfo"/>
+                <xs:attribute name="name" type="xs:string" use="required" hpcc:displayName="Name"
+                              hpcc:autoGenerateType="prefix" hpcc:autoGenerateValue="ws_sql_service"
```

Comment:
do these "hpcc:autoGenerate" attributes work in conjunction?  Does this setting mean that we will automatically populate the name of a new instance as "ws_sql_service"+"something"?

Reply:
Yes. For reference, see http://10.240.34.10/wiki/index.php/XSD_Schema_Definition#Automatic_Value_Generation

Reply:
very good. what are the other autoGenerateTypes? if "prefix" is selected what is appended to the prefix? a counter? Let's consider how the 2 attributes can be combined.

Reply:
@kenrowland back to you.

Reply:
Is the information on the Wiki not sufficient? If so, let me know what is not clear and I will make a change there. As for combining the two attributes (autoGenerateType and autoGenerateValue) that should be done with a separate issue since it affects a number of XSDs. If it was changed, it would look like autoGenerate=type,data. Please open a JIRA if you would like these combined into a single attribute.

Reply:
https://track.hpccsystems.com/browse/HPCC-21132
Also, please update wiki to clearly convey all options available to hpcc::autoGenerateType
the top level description should have, hpcc::autoGenerateType"prefix|prefix_|prefix#|configProperty|fixedValue|siblingValue"
then you can break down, define, and group the options as appropriate @kenrowland

Reply:
I will make changes to the Wiki based on the outcome of 21132 (which will affect this and other XSDs). Unless there are other changes needed for this PR, please mark approved. @rpastrana